tldw_Server_API/app/core/LLM_Calls/routing/candidate_pool.py

```
@@ -0,0 +1,198 @@
+        ),
+        metadata=metadata,
+    )
+
+
+def _as_candidate(record: RoutingCandidate | Mapping[str, Any]) -> RoutingCandidate:
+    if isinstance(record, RoutingCandidate):
+        return record
+    return _candidate_from_mapping(record)
+
+
+def _matches_capabilities(
```

Ruff (0.15.6)

[warning] Too many return statements (7 > 6)

(PLR0911)